crates/core/src/network/net_util.rs

```diff
@@ -20,42 +18,48 @@ use hyper::body::{Body, Incoming};
 use hyper_util::rt::TokioIo;
 use hyper_util::server::graceful::GracefulShutdown;
 use tokio::io;
-use tokio::net::{TcpListener, UnixListener, UnixStream};
-use tokio_util::net::Listener;
+use tokio::net::UnixStream;
+use tokio_util::either::Either;
 use tonic::transport::{Channel, Endpoint};
 use tracing::{Instrument, Span, debug, error_span, info, instrument, trace};
 
 use restate_types::config::Configuration;
 use restate_types::errors::GenericError;
+use restate_types::net::address::{AdvertisedAddress, GrpcPort};
+use restate_types::net::address::{ListenerPort, PeerNetAddress};
 use restate_types::net::connect_opts::CommonClientConnectionOptions;
-use restate_types::net::{AdvertisedAddress, BindAddress};
+use restate_types::net::listener::Listeners;
 
 use crate::{ShutdownError, TaskCenter, TaskKind, cancellation_watcher};
 
-pub fn create_tonic_channel<T: CommonClientConnectionOptions + Send + Sync + ?Sized>(
-    address: AdvertisedAddress,
+pub fn create_tonic_channel<
```

Comment:
nit + possible follow-up: Maybe this method and `create_channel` in `grpc/connector.rs` could be merged. They look quite similar.